commons-component-upgrade/src/test/java/org/exoplatform/commons/version/util/VersionComparatorTest.java

```diff
@@ -11,6 +11,28 @@ public void testIsBefore() {
     assertFalse(VersionComparator.isBefore("2.1", ""));
     assertTrue(VersionComparator.isBefore("", "2.2"));
     assertTrue(VersionComparator.isBefore("5.0.0-M32", "5.0-RC1"));
+    assertTrue(VersionComparator.isBefore("6.2.0-20210529", "6.2.0-20210531"));
+    assertFalse(VersionComparator.isBefore("6.2.0-20210601", "6.2.0-20210531"));
+    assertTrue(VersionComparator.isBefore("6.1.1", "6.2.0-20210531"));
+    assertFalse(VersionComparator.isBefore("6.2.1", "6.2.0-20210531"));
+    assertFalse(VersionComparator.isBefore("6.2.0", "6.2.0-20210531"));
+    assertTrue(VersionComparator.isBefore("6.2.0-20210529", "6.2.0-20210531"));
+    assertFalse(VersionComparator.isBefore("6.2.0-20210601", "6.2.0-20210531"));
+    assertTrue(VersionComparator.isBefore("6.2.1-20210529", "6.2.1-20210531"));
+    assertFalse(VersionComparator.isBefore("6.2.1-20210601", "6.2.1-20210531"));
+    assertTrue(VersionComparator.isBefore("6.1.0", "6.2.0-20210531"));
+    assertTrue(VersionComparator.isBefore("6.1.1", "6.2.0-20210531"));
+    assertFalse(VersionComparator.isBefore("6.2.0", "6.2.0-M20"));
+    assertFalse(VersionComparator.isBefore("6.2.0", "6.2.0-20210531"));
+    assertTrue(VersionComparator.isBefore("6.2.0-20210531", "6.2.0"));
+    assertFalse(VersionComparator.isBefore("6.2.1", "6.2.0-20210531"));
+    assertFalse(VersionComparator.isBefore("6.2.1", "6.2.0-20210531"));
+    assertTrue(VersionComparator.isBefore("6.2.0-20210531", "6.2.0"));
+    assertTrue(VersionComparator.isBefore("6.2.0-20210531", "6.2.1"));
+    assertTrue(VersionComparator.isBefore("6.2.0-20210531", "6.2.0-m01"));
+    assertTrue(VersionComparator.isBefore("6.2.0-20210531", "6.2.0-rc01"));

```

Comment:
I'm not sure of theses 2 cases.
6.2.0-M25 can be released on 20210601.
Then it will be after 6.2.0-20210531 but before 6.2.0-20210602. Same case for RC. How do we resolved this case ?

Reply:
As mentioned in description, The milestones will be considered always after the CI/CD versions. In general case, either we deploy milestones on a host or use CI/CD versions.

Reply:
ok. Thanks for the clarification